packages/shared/isValidElementType.js

```diff
@@ -30,7 +30,8 @@ export default function isValidElementType(type: mixed) {
     type === REACT_PLACEHOLDER_TYPE ||
     (typeof type === 'object' &&
       type !== null &&
-      (type.$$typeof === REACT_PROVIDER_TYPE ||
+      (typeof type.then === 'function' ||
```

Comment:
Let's put the `.then` comparison last since it is the least common one.

Reply:
Is it really? I intentionally put it first because it seems like it will be the most common one. Providers are pretty rare. Perhaps ForwardRef will more common, in the near term.

Reply:
ok, that's fair.